libraries/AP_HAL_Linux/RCInput_ZYNQ.cpp

```diff
@@ -45,13 +45,17 @@ void RCInput_ZYNQ::init()
 void RCInput_ZYNQ::_timer_tick()
 {
     uint32_t v;
-
     // all F's means no samples available
     while((v = *pulse_input) != 0xffffffff) {
-        // Hi bit indicates pin state, low bits denote pulse length
-        if(!(v & 0x80000000))
-            _s0_time = (v & 0x7fffffff)/TICK_PER_US;
-        else
-            _process_rc_pulse(_s0_time, (v & 0x7fffffff)/TICK_PER_US);
+        // This calculation is done everytime
+        auto temp = (v & 0x7fffffff)/TICK_PER_US;
+        // Hi bit indicates low pin state
+        // Lower 31 bits denote pulse length
+        if (v & 0x80000000) {
```

Comment:
Just to be sure, this is the opposite of what was in the current driver, right? Can you add some explanation for this change in your original PR comment?